tests/manage/pv_services/pvc_snapshot/test_encrypted_rbd_pvc_snapshot.py

```diff
@@ -0,0 +1,396 @@
+    ],
+)
+@tier1
+@skipif_ocs_version("<4.8")
+@skipif_ocp_version("<4.8")
+class TestEncryptedRbdBlockPvcSnapshot(ManageTest):
+    """
+    Tests to take snapshots of encrypted RBD Block VolumeMode PVCs
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, kv_version, request):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        # Initialize Vault
+        self.vault = kms.Vault()
+        self.vault.gather_init_vault_conf()
+        self.vault.update_vault_env_vars()
+
+        # Check if cert secrets already exist, if not create cert resources
+        ocp_obj = OCP(kind="secret", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        try:
+            ocp_obj.get_resource(resource_name="ocs-kms-ca-secret", column="NAME")
+        except CommandFailed as cfe:
+            if "not found" not in str(cfe):
+                raise
+            else:
+                self.vault.create_ocs_vault_cert_resources()
+
+        # Create vault namespace, backend path and policy in vault
+        self.vault_resource_name = create_unique_resource_name("test", "vault")
+        self.vault.vault_create_namespace(namespace=self.vault_resource_name)
+        self.vault.vault_create_backend_path(
+            backend_path=self.vault_resource_name, kv_version=kv_version
+        )
+        self.vault.vault_create_policy(policy_name=self.vault_resource_name)
+
+        ocp_obj = OCP(kind="configmap", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+
+        # If csi-kms-connection-details exists, edit the configmap to add new vault config
+        try:
+            ocp_obj.get_resource(
+                resource_name="csi-kms-connection-details", column="NAME"
+            )
+            self.new_kmsid = self.vault_resource_name
+            vdict = VAULT_CSI_CONNECTION_CONF
+            for key in vdict.keys():
+                old_key = key
+            vdict[self.new_kmsid] = vdict.pop(old_key)
+            vdict[self.new_kmsid]["VAULT_BACKEND_PATH"] = self.vault_resource_name
+            vdict[self.new_kmsid]["VAULT_NAMESPACE"] = self.vault_resource_name
+            kms.update_csi_kms_vault_connection_details(vdict)
+
+        except CommandFailed as cfe:
+            if "not found" not in str(cfe):
+                raise
+            else:
+                self.new_kmsid = "1-vault"
+                self.vault.create_vault_csi_kms_connection_details()
+
+        def finalizer():
+            # Remove the vault config from csi-kms-connection-details configMap
+            if len(kms.get_encryption_kmsid()) > 1:
+                kms.remove_kmsid(self.new_kmsid)
+
+            # Delete the resources in vault
+            self.vault.remove_vault_backend_path()
+            self.vault.remove_vault_policy()
+            self.vault.remove_vault_namespace()
+
+        request.addfinalizer(finalizer)
+
+    def test_encrypted_rbd_block_pvc_snapshot(
```

Comment:
The test case's implementation has some similarities to tests.manage.pv_services.test_rbd_pv_encryption.TestRbdPvEncryption.test_rbd_pv_encryption

Please try to merge the similarities into functions, which will be called from both tests

Reply:
Done